pkg/codegen/codegen_test.go

```diff
@@ -307,3 +309,56 @@ components:
           type: string
           enum: [car, dog, oldage]
 `
+
+func TestGenerateEnumTypes(t *testing.T) {
+	tests := []struct {
+		name    string
+		types   []TypeDefinition
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "string",
+
+			types: []TypeDefinition{{
+				JsonName: "my_type",
+				TypeName: "MyType",
+				Schema: Schema{
+					GoType:     "string",
+					RefType:    "string",
+					EnumValues: map[string]string{"some": "value"},
+				},
+			}},
+			want: "\n// MyType defines model for my_type.\ntype MyType struct {\n    value string\n}\nfunc (t MyType) ToValue() string {\n    return t.value\n}\nfunc (t MyType) MarshalJSON() ([]byte, error) {\n    return json.Marshal(t.value)\n}\nfunc (t MyType) UnmarshalJSON(data []byte) error {\n    var value string\n    if err := json.Unmarshal(data, &value); err != nil {\n        return err\n    }\n    return nil\n}\nfunc (t MyType) FromValue(value string) error {\n    switch value {\n    \n    case some.value:\n        t.value = value\n        return nil\n    \n    }\n    return fmt.Errorf(\"unknown enum value: %v\", value)\n}",
```

Comment:
Might be more readable here to use backtick strings to make the expected value more readable in source code

Reply:
Backticks look awful because of the indentation.